src/clocks.h

```diff
@@ -1,108 +1,272 @@
+ *
+ * - Hibernation/suspend time is included
+ * - adjtime()/NTP adjustments affect this clock
+ *
+ * @return Time since EPOCH
+ */
 extern usec_t now_realtime_usec(void);
 
+/**
+ * Returns time from monotonic clock.
+ *
+ * If monotonic clock is available:
+ * - hibernation/suspend time is not included
+ * - adjtime()/NTP adjusments affect this clock
+ * If monotonic clock is not available, this fallbacks to `now_realtime()`.
+ *
+ * @param tv struct timeval to update
+ * @return time
+ */
 extern int now_monotonic_timeval(struct timeval *tv);
+/**
+ * Returns time from monotonic clock.
+ *
+ * If monotonic clock is available:
+ * - hibernation/suspend time is not included
+ * - adjtime()/NTP adjusments affect this clock
+ * If monotonic clock is not available, this fallbacks to `now_realtime()`.
+ *
+ * @return time
+ */
 extern time_t now_monotonic_sec(void);
+/**
+ * Returns time from monotonic clock.
+ *
+ * If monotonic clock is available:
+ * - hibernation/suspend time is not included
+ * - adjtime()/NTP adjusments affect this clock
+ * If monotonic clock is not available, this fallbacks to `now_realtime()`.
+ *
+ * @return time
+ */
 extern usec_t now_monotonic_usec(void);
 
+/**
+ * Returns time from boottime clock
+ * 
+ * If boottime clock is available:
+ * - hibernation/suspend time is included
+ * - adjtime()/NTP adjusments affect this clock
+ * If boottime clock is not available, this fallbacks to now_monotonic().
+ * If monotonic clock is not available, this fallbacks to `now_realtime()`.
+ *
+ * @param tv struct timeval to update.
+ * @return time
+ */
 extern int now_boottime_timeval(struct timeval *tv);
+/**
+ * Returns time from boottime clock
+ * 
+ * If boottime clock is available:
+ * - hibernation/suspend time is included
+ * - adjtime()/NTP adjusments affect this clock
+ * If boottime clock is not available, this fallbacks to now_monotonic().
+ * If monotonic clock is not available, this fallbacks to `now_realtime()`.
+ *
+ * @return time
+ */
 extern time_t now_boottime_sec(void);
+/**
+ * Returns time from boottime clock
+ * 
+ * If boottime clock is available:
+ * - hibernation/suspend time is included
+ * - adjtime()/NTP adjusments affect this clock
+ * If boottime clock is not available, this fallbacks to now_monotonic().
+ * If monotonic clock is not available, this fallbacks to `now_realtime()`.
+ *
+ * @return time
+ */
 extern usec_t now_boottime_usec(void);
 
-
+/**
+ * Convert `struct timveal` to `usec_t`
+ *
+ * @param ts struct timeval
+ * @return time in microseconds 
+ */
 extern usec_t timeval_usec(struct timeval *ts);
+/**
+ * Return microseconds between `now` and `old`.
+ *
+ * This returns
+ * ~~~~~~~~~~~~~~{.c}
+ * abs(now - old)
+ * ~~~~~~~~~~~~~~
+ *
+ * @param now first struct timeval
+ * @param old second struct timeval
+ * @return time in microseconds between `now` and `old`
+ */
 extern usec_t dt_usec(struct timeval *now, struct timeval *old);
 
+/**
+ * Reset heartbeat to 0LL.
```

Comment:
heartbeat_t is used as an opaque type. It could very well become a struct if needed (for example to store the offset with realtime clock in order to enhance alignment as suggested by @ktsaou). Therefore I don't think we should describe here a specific value (this can be done in the function definition if needed).